.circleci/config.yml

```diff
@@ -0,0 +1,53 @@
+version: 2.1
+
+parameters:
+  ubuntu_focal_machine:
+    type: string
+    default: 'ubuntu-2004:202111-02'
+  ubuntu_xenial_machine:
+    type: string
+    default: 'ubuntu-1604:202104-01'
+
+workflows:
+  circleci_build_and_test:
+    jobs:
+      - spin_up:
+          name: 'spin_up_<< matrix.config >>_<< matrix.machine >>'
+          matrix:
+            parameters:
+              config: ['', 'betanet', 'testnet', 'devnet', 'dev', 'beta', 'release', 'nightly']
+              machine: [<< pipeline.parameters.ubuntu_focal_machine >>, << pipeline.parameters.ubuntu_xenial_machine >>]
+
+jobs:
+  spin_up:
+    parameters:
+      config:
+        type: string
+      machine:
+        type: string
+    machine:
+      image: << parameters.machine >>
+    resource_class: medium
+    steps:
+      - checkout
+      - run:
+          name: Install docker
+          command: |
+            export SUDO=sudo
+            $SUDO apt update
+            $SUDO apt -y install ca-certificates curl gnupg lsb-release
+            $SUDO curl -fsSL https://download.docker.com/linux/ubuntu/gpg | $SUDO gpg --dearmor -o /usr/share/keyrings/docker-archive-keyring.gpg
+            $SUDO echo "deb [arch=$(dpkg --print-architecture) signed-by=/usr/share/keyrings/docker-archive-keyring.gpg] https://download.docker.com/linux/ubuntu \
+              $(lsb_release -cs) stable" | $SUDO tee /etc/apt/sources.list.d/docker.list > /dev/null
+            $SUDO apt update
+            $SUDO apt -y install docker-ce docker-ce-cli containerd.io
+      - run:
+          name: CircleCI docker download fix
+          command: sudo iptables -I INPUT -m conntrack --ctstate INVALID -j DROP
+      - run:
+          name: Stand up sandbox
+          command: |
+            export CONFIG=<< parameters.config >>
+            set -e
+            if [ -n "$DOCKER_PASSWORD" ]; then echo "$DOCKER_PASSWORD" | docker login -u "$DOCKER_USERNAME" --password-stdin ; fi
```

Comment:
I think we can remove this line altogether, unless you're using a private repo.

Reply:
I think that the login helps performance when using docker to pull images.

Reply:
done